pkg/registry/service/ipallocator/allocator.go

```diff
@@ -107,7 +107,7 @@ func (r *Range) Free() int {
 func (r *Range) Allocate(ip net.IP) error {
 	ok, offset := r.contains(ip)
 	if !ok {
-		return ErrNotInRange
+		return errors.New(fmt.Sprintf("%v %v", r, ip))
```

Comment:
If you need to return additional information, make an object implementing error. People are actually checking for the ErrNotInRange and this change will break them.